lib/src/models/custom_item.dart

```diff
@@ -0,0 +1,52 @@
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_login/src/widgets/animated_text_form_field.dart';
+
+abstract class ItemBuilder {
+  Widget build(BuildContext context, double textFieldWidth, List<dynamic> datas,
+      int index);
+  String get content;
+}
+
+class PureTextItem extends ItemBuilder {
```

Comment:
I don't think the name of the class is descriptive enough, maybe we can use something like `LoginTextField`

Reply:
It can be both used in Sign Up and Login.So I don't think it is good to name it after Login.

Reply:
Hmm and something like AuthTextField?